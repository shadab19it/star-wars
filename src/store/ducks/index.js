import { combineReducers } from "redux";

// import login from "./login";
// import input from "./login";
import films from "./films";
import filmList from "./filmList";

export default combineReducers({
    // login,
    films,
    filmList
    // input
});
