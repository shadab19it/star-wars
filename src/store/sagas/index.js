import { all } from "redux-saga/effects";
import { getFilmsWatcher } from "./films";

export default function* rootSaga() {
    yield all([getFilmsWatcher()]);
}
