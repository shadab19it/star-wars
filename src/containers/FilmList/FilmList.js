import React, { Component, Fragment } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Container } from "react-bootstrap";
import { Creators as List } from "../../store/ducks/filmList";
import Film from "../../components/Film/Film";
import "./FilmList.scss";

class FilmList extends Component {
    constructor(props) {
        super(props);

        this.state = {
            error: false,
            isLoading: false,
            list: []
        };
    }

    componentDidMount() {
        this.props.filmList.cleanFilms();
        this.props.getFilms();

        this.loadList();
    }

    componentDidUpdate() {
        if (
            this.props.films &&
            this.props.films.results &&
            this.props.films.results.length &&
            this.props.films.results[0] === "error" &&
            !this.state.error
        ) {
            this.setState({
                error: "Erro1",
                isLoading: false
            });
        } else if (
            this.props.films &&
            this.props.films.results &&
            this.props.films.results.length &&
            !this.state.list.length
        ) {
            this.setState({
                list: this.props.films.results,
                isLoading: false
            });
        }
    }

    loadList = () => {
        this.setState({ isLoading: true });
        this.props.getFilms();
    };

    removeFilm = id => {
        this.props.removeFilm(id);

        this.setState({
            list: this.state.list.filter(function(item) {
                return item.id !== id;
            })
        });
    };

    editFilm = id => {
        this.props.history.push(`/film/${id}`);
    };

    render() {
        const { error, isLoading, list } = this.state;

        return (
            <Container className="text-center film-list-container">
                <div className="row film-list-header">
                    <div className="col-3" key="Nome">
                        <span>Nome</span>
                    </div>
                    <div className="col-3" key="Diretor">
                        <span>Diretor</span>
                    </div>
                    <div className="col-3" key="DataDeLançamento">
                        <span>Data de lançamento</span>
                    </div>
                    <div className="col-3" key="Detalhes">
                        <span>Detalhes</span>
                    </div>
                </div>
                {!error &&
                    !isLoading &&
                    list.map((film, i) => {
                        return (
                            <Fragment key={i}>
                                <Film
                                    data={film}
                                    status="idStatusConciliacao"
                                    removeFilm={this.removeFilm}
                                    editFilm={this.editFilm}
                                />
                            </Fragment>
                        );
                    })}
                {error && (
                    <div>
                        Erro ao buscar os filmes. Verifique se tem conexão com a
                        internet e tente novamente mais tarde.
                    </div>
                )}
                {isLoading && (
                    <div>
                        <p className="text-3 text-color-primary my-0 pt-1">
                            Carregando...
                        </p>
                    </div>
                )}
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        filmList: state.filmList,
        films: state.films
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getFilms: () => dispatch({ type: "GET_FILMS" }),
        removeFilm: id =>
            dispatch({
                type: "REMOVE_FILM",
                id: id
            }),
        filmList: bindActionCreators(List, dispatch)
    };
};

export const FilmListWithoutWithRouter = connect(
    mapStateToProps,
    mapDispatchToProps
)(FilmList);

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(FilmList)
);
