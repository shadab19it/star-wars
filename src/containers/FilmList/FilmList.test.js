import React from "react";
import { FilmListWithoutWithRouter } from "./FilmList";
import { IntlProvider } from "react-intl";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../store";

configure({ adapter: new Adapter() });

const country = "pt-BR";

describe("FilmList Container", () => {
    const component = (
        <IntlProvider locale={country}>
            <Provider store={store}>
                <FilmListWithoutWithRouter />
            </Provider>
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated a film list with a header inside", () => {
        const wrapper = mount(component);
        const header = wrapper.find(".film-list-header");
        expect(header.length).toEqual(1);
    });

    it("generated an film list that is loading films", () => {
        const wrapper = mount(component);
        const loading = wrapper
            .find(".container div")
            .last()
            .text();
        expect(loading).toBe("Carregando...");
    });
});
