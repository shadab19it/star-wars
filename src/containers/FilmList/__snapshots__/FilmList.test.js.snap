// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList Container matches the snapshot 1`] = `
<div
  className="text-center film-list-container container"
>
  <div
    className="row film-list-header"
  >
    <div
      className="col-3"
    >
      <span>
        Nome
      </span>
    </div>
    <div
      className="col-3"
    >
      <span>
        Diretor
      </span>
    </div>
    <div
      className="col-3"
    >
      <span>
        Data de lan√ßamento
      </span>
    </div>
    <div
      className="col-3"
    >
      <span>
        Detalhes
      </span>
    </div>
  </div>
  <div>
    <p
      className="text-3 text-color-primary my-0 pt-1"
    >
      Carregando...
    </p>
  </div>
</div>
`;
